src/observable/ScalarObservable.ts

```diff
@@ -80,8 +80,8 @@ proto.reduce = function <T, R>(project: (acc: R, x: T) => R, seed?: R): Observab
     return <any>this;
   }
   let result = tryCatch(project)(seed, this.value);
-  if (result === errorObject) {
-    return new ErrorObservable(errorObject.e);
+  if (result as any === errorObject) {
```

Comment:
Oh, I prefer `result as any` but I think `<any>result` is more used throughout the codebase.

Regarding the use of casting in this case, changing the signature of [tryCatch()](https://github.com/ReactiveX/RxJS/blob/master/src/util/tryCatch.ts) to return something like `(...args: any[]) => any` instead of just `Function` solves the problem?